### PR TITLE
[WIP] Provisioning wrapper for ceph-installer

### PR DESCRIPTION
**Pull request by nthomas-redhat**
tendrl-bug-id:Tendrl/node-agent/issues#202

Signed-off-by: Nishanth Thomas <nthomas@redhat.com>

### CODE REVIEW

File: tendrl/node_agent/provisioner/ceph/plugins/ceph_installer.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    _CEPH_INSTALLER_API_PORT = '8181'
+
+    def __init__(
+        self,
+        provisioner_node
+    ):
+        self.http = urllib3.PoolManager()
+        self.provisioner_node = provisioner_node
+
+    def install_mon(self, mons):
```
Comment (shtripat) [inline tendrl/node_agent/provisioner/ceph/plugins/ceph_installer.py]: In install_mon and install_osd, only method name differs I feel. Can we have a common function called from both?

File: tendrl/node_agent/provisioner/ceph/manager.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import importlib
+import inspect
+import logging
+import os
+import re
+
+from tendrl.node_agent.provisioner.ceph import provisioner
```
Comment (r0h4n) [inline tendrl/node_agent/provisioner/ceph/manager.py]: This and L15 can cause scope issues for "provisioner", please fix

File: tendrl/node_agent/provisioner/ceph/manager.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+LOG = logging.getLogger(__name__)
+
+
+class ProvisioningManager(object):
+
+    def __init__(self, provisioner):
+        self.ceph_provisioner = provisioner
```
Comment (r0h4n) [inline tendrl/node_agent/provisioner/ceph/manager.py]: same as #L7

File: tendrl/node_agent/provisioner/ceph/manager.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        try:
+            path = os.path.dirname(os.path.abspath(__file__)) + '/plugins'
+            pkg = 'tendrl.node_agent.provisioner.ceph.plugins'
+            for py in [f[:-3] for f in os.listdir(path)
+                       if f.endswith('.py') and f != '__init__.py']:
+                plugin_name = '.'.join([pkg, py])
+                mod = importlib.import_module(plugin_name)
+                clsmembers = inspect.getmembers(mod, inspect.isclass)
+                for name, cls in clsmembers:
+                    exec("from %s import %s" % (plugin_name, name))
```
Comment (r0h4n) [inline tendrl/node_agent/provisioner/ceph/manager.py]: exec is not safe, please use something like https://github.com/Tendrl/alerting/blob/5a84e02bf060a54f9826b0850f819a86af5642ef/tendrl/alerting/handlers/__init__.py#L118

**Pull request closed without merging**